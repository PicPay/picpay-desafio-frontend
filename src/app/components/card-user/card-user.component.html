<div class="user" (click)="paymentUser(user)">
    <img src="{{user.img}}" alt="Avatar do usÃºario"/>
    <div class="info-wrapper">
        <div>
            {{user.username}}
        </div>
        <div class="user-name">
            <small>{{user.name}}</small>
        </div>
    </div>
</div>