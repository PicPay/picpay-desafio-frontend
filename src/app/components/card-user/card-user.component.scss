/* List of Users */
.user {
    box-shadow: 0 1px 0 #ffffff, 0px -1px 0px #ffffff;
    padding: 0.5rem;
    display: flex;
    align-items: center;

    &:hover{
        background-color: lightgrey;
        cursor: pointer;
    }
}

.user-name {
    opacity: 0.7;
}

.user > img {
    border-radius: 50%;
    height: 4rem;
    width: 4rem;
    border: 1px solid lightgrey;
}

.user > .info-wrapper {
    margin-left: 1rem;
}