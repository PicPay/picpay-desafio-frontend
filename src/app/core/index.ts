export * from './components/index';
export * from './model/index';
export * from './services/index';
