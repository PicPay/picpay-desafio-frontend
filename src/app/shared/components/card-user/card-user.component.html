<div *ngIf="user" class="card" [routerLink]="['/users', user.id]">
  <img class="card__image" [src]="user.img" alt="User image" />
  <div class="card__info">
    <span class="card__name card__info--truncated">{{ user.name }}</span>
    <small class="card__info--truncated">{{ user.username }}</small>
  </div>
</div>

<div *ngIf="!user" class="card">
  <ngx-skeleton-loader
    [theme]="{
      width: '70px',
      height: '70px',
      margin: '0',
      'margin-right': '10px',
      'border-radius': '10px'
    }"
  ></ngx-skeleton-loader>
  <div class="card__info">
    <ngx-skeleton-loader
      [theme]="{
        width: '150px',
        margin: '0'
      }"
    ></ngx-skeleton-loader>
    <ngx-skeleton-loader
      [theme]="{
        width: '80px',
        margin: '0'
      }"
    ></ngx-skeleton-loader>
  </div>
</div>
