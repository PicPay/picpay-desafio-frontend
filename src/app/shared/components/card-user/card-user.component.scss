@import '~src/sass/variables';

.card {
    display: flex;
    align-items: center;
    padding: 20px;
    background-color: #FAFAFA;
    cursor: pointer;
    border-radius: 20px;
    margin: 10px;

    &:hover {
        -webkit-box-shadow: 0 0 10px 0 rgb(0 0 0 / 10%);;
        -moz-box-shadow: 0 0 10px 0 rgb(0 0 0 / 10%);;
        box-shadow: 0 0 10px 0 rgb(0 0 0 / 10%);
    }

    &__image {
        width: 70px;
        height: 70px;
        border-radius: 10px;
        margin-right: 10px;
    }

    &__info {
        flex: 1;
        display: flex;
        flex-direction: column;
        white-space: nowrap;
        overflow: hidden;

        &--truncated {
            overflow: hidden;
            text-overflow: ellipsis;
        }
    }

    &__name {
        font-weight: bold;
    }
}