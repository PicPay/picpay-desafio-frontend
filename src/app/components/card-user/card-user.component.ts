import { Component, Input, OnInit, TemplateRef, ViewChild } from '@angular/core';
import { TransactionResponse } from '../../models/transaction-response.interface';
import { ModalService } from '../../services/modal.service';
import { User } from '../../models/user.interface';
import { FormPaymentComponent } from '../form-payment/form-payment.component';

@Component({
  selector: 'app-card-user',
  templateUrl: './card-user.component.html',
  styleUrls: ['./card-user.component.scss']
})
export class CardUserComponent implements OnInit {

  @Input()
  user: User;

  @ViewChild('paymentMessageResponse', { static: true })
  private paymentMessageResponseTpl: TemplateRef<any>;

  paymentStatus: 'Aprovada' | 'Reprovada';

  constructor(
    private modalService: ModalService,
  ) { }

  ngOnInit() {
  }

  public payUserModal(user: User): void {
    const modalTitle = `Pagamento para ${user.name}`
    const modalRef = this.modalService.open(FormPaymentComponent, modalTitle, user);

    modalRef.afterClosed$.subscribe((res) => {
      if(res && res.data) {
        this.showModalPaymentMessageResponse(res.data);
      }
    });
  }

  public showModalPaymentMessageResponse(transaction: TransactionResponse): void {
    this.paymentStatus = transaction.status;
    this.modalService.open(
      this.paymentMessageResponseTpl,
      'Recibo de pagamento',
      null
    );
  }

}
