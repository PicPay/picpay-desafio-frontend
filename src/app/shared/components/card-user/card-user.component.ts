import { Component, Input } from '@angular/core';
import { User } from '@shared/interfaces/user';

@Component({
  selector: 'ngx-card-user',
  templateUrl: 'card-user.component.html',
  styleUrls: ['card-user.component.scss'],
})
export class CardUserComponent {
  @Input() user: User;
}
