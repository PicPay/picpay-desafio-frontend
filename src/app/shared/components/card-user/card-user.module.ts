import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CardUserComponent } from './card-user.component';

@NgModule({
  declarations: [CardUserComponent],
  exports: [CardUserComponent],
  imports: [CommonModule],
})
export class CardUserModule {}
