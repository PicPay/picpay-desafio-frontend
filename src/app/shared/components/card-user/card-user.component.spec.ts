import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { CardUserComponent } from './card-user.component';
import { CardUserModule } from './card-user.module';

describe('CardUserComponent', () => {
  let component: CardUserComponent;
  let fixture: ComponentFixture<CardUserComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, CardUserModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CardUserComponent);
    component = fixture.componentInstance;

    component.user = {
      id: 1,
      name: 'Test 1',
      username: '@test1',
      img: 'http://test1.png',
    };

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should create snapshot with user', () => {
    expect(fixture).toMatchSnapshot('with_user');
  });

  it('should create snapshot without user', () => {
    component.user = null;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot('without_user');
  });
});
