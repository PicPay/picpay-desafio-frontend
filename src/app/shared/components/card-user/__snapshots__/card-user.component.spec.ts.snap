// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardUserComponent should create snapshot with user: with_user 1`] = `
<ngx-card-user
  user={[Function Object]}
>
  <div
    class="card"
    tabindex="0"
  >
    <img
      alt="User image"
      class="card__image"
      src="http://test1.png"
    />
    <div
      class="card__info"
    >
      <span
        class="card__name card__info--truncated"
      >
        Test 1
      </span>
      <small
        class="card__info--truncated"
      >
        @test1
      </small>
    </div>
  </div>
</ngx-card-user>
`;

exports[`CardUserComponent should create snapshot without user: without_user 1`] = `
<ngx-card-user
  user="null"
>
  <div
    class="card"
  >
    <ngx-skeleton-loader>
      
      <span
        aria-busy="true"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuetext="Loading..."
        class="loader progress"
        role="progressbar"
        style="width: 70px; height: 70px; margin: 0px 10px 0px 0px; border-radius: 10px;"
        tabindex="0"
      />
    </ngx-skeleton-loader>
    <div
      class="card__info"
    >
      <ngx-skeleton-loader>
        
        <span
          aria-busy="true"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuetext="Loading..."
          class="loader progress"
          role="progressbar"
          style="width: 150px; margin: 0px;"
          tabindex="0"
        />
      </ngx-skeleton-loader>
      <ngx-skeleton-loader>
        
        <span
          aria-busy="true"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuetext="Loading..."
          class="loader progress"
          role="progressbar"
          style="width: 80px; margin: 0px;"
          tabindex="0"
        />
      </ngx-skeleton-loader>
    </div>
  </div>
</ngx-card-user>
`;
